shipyard_blueprints/ftp/shipyard_ftp/cli/download.py

```diff
@@ -171,126 +58,88 @@ def find_matching_files(file_names, file_name_re):
-def main():
-    args = get_args()
-    host = args.host
-    port = args.port
-    username = args.username
-    password = args.password
-    source_file_name = args.source_file_name
-    source_folder_name = clean_folder_name(args.source_folder_name)
-    source_full_path = combine_folder_and_file_name(
-        folder_name=source_folder_name, file_name=source_file_name
-    )
-    source_file_name_match_type = args.source_file_name_match_type
+        destination_folder_name = shipyard.clean_folder_name(
+            args.destination_folder_name
+        )
+        if destination_folder_name:
+            shipyard.create_folder_if_dne(destination_folder_name)
+        client = FtpClient(host=host, port=port, user=username, pwd=password)
 
-    destination_folder_name = clean_folder_name(args.destination_folder_name)
-    if not os.path.exists(destination_folder_name) and (destination_folder_name != ""):
-        os.makedirs(destination_folder_name)
+        logger.info("Beginning the download process...")
+        if source_file_name_match_type == "exact_match":
+            try:
+                destination_name = shipyard.determine_destination_full_path(
+                    destination_folder_name=destination_folder_name,
+                    destination_file_name=args.destination_file_name,
+                    source_full_path=source_full_path,
+                )
+                client.download(source_full_path, destination_name)
 
-    client = get_client(host=host, port=port, username=username, password=password)
+            except Exception as e:
+                logger.error(
+                    f"Failed to download {source_full_path} due to {e}\n"
+                    f"Most likely, the file name/folder name you specified has typos or the full folder name was "
+                    f"not provided. Check these and try again."
+                )
+                raise e
+        elif source_file_name_match_type == "regex_match":
+            folders = [source_folder_name]
+            files = []
+            while folders:
+                folder_filter = folders[0]
+                files, folders = client.find_files_in_directory(
+                    folder_filter=folder_filter, files=files, folders=folders
+                )
 
-    print("Beginning the download process...")
-    if source_file_name_match_type == "regex_match":
-        folders = [source_folder_name]
-        files = []
-        while folders != []:
-            folder_filter = folders[0]
-            files, folders = find_files_in_directory(
-                client=client, folder_filter=folder_filter, files=files, folders=folders
+            matching_file_names = shipyard.find_all_file_matches(
+                files, re.compile(source_file_name)
             )
 
-        matching_file_names = find_matching_files(files, re.compile(source_file_name))
-
-        number_of_matches = len(matching_file_names)
+            number_of_matches = len(matching_file_names)
```

Comment:
same comment as above using the `:=` operator inline with the `if` statement. Not a requirement, I just think it adds a bit of syntactic sugar